core/server/models/user.js

```diff
@@ -612,11 +612,17 @@ User = ghostBookshelf.Model.extend({
 
         return self.forge({id: userId}).fetch({require: true}).then(function then(_user) {
             user = _user;
-            if (userId === options.context.user) {
+            if (Number(userId) === options.context.user) {
+                // If the old password wasn't sent through, make it blank
+                if (!oldPassword) {
+                    oldPassword = '';
+                }
+                // Check the old password entered against the user's old password
                 return bcryptCompare(oldPassword, user.get('password'));
+            } else {
```

Comment:
Do we really need the else statement here? As the line above is returning the result of bcryptCompare and thus exiting the function, the extra else IMO isn't needed :smile: